src/coreclr/tests/src/profiler/unittest/getappdomainstaticaddress.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.IO;
+using System.Runtime.Loader;
+
+namespace Profiler.Tests
+{
+    class Program
+    {
+        static readonly Guid GetAppDomainStaticAddressProfilerGuid = new Guid("604D76F0-2AF2-48E0-B196-80C972F6AFB7");
+
+        static int Main(string[] args)
+        {
+            if (args.Length == 1 && args[0].Equals("RunTest", StringComparison.OrdinalIgnoreCase))
+            {
+                return RunTest();
+            }
+
+            return ProfilerTestRunner.Run(profileePath: System.Reflection.Assembly.GetExecutingAssembly().Location,
+                                          testName: "UnitTestGetAppDomainStaticAddress",
+                                          profilerClsid: GetAppDomainStaticAddressProfilerGuid);
+        }
+
+        static int RunTest()
+        {
+            LoadCollectibleAssembly();
+
+            Thread.Sleep(TimeSpan.FromSeconds(3));
```

Comment:
There is no guarantee that the collectible context is going to be collected here. If you want to be sure it was collected, you'd need to make the LoadCollectibleAssembly non-inlineable, return a weak reference to the dynamicLibrary from it and then add calling GC.Collect / GC.WaitForPendingFinalizers in a loop while the weak reference is alive. See https://docs.microsoft.com/en-us/dotnet/standard/assembly/unloadability.